tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,602 @@
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("%s (reason 0x%02x)", addr, reason);
+
+	if (default_conn != conn) {
+		FAIL("Conn mismatch disconnect %s %s)", default_conn, conn);
+		return;
+	}
+
+	bt_conn_unref(default_conn);
+	default_conn = NULL;
+	UNSET_FLAG(is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void send_sdu(int iteration, int chan_idx, int bytes)
+{
+	struct bt_l2cap_chan *chan = &channels[chan_idx].le.chan;
+	struct net_buf *buf;
+
+	/* First 4 bytes in sent payload is iteration count */
+	sys_put_le32(iteration, channels[chan_idx].payload);
+
+	if (channels[chan_idx].buf != 0) {
+		FAIL("Buf should have been dealocated by now");
+		return;
+	}
+
+	if (chan_idx == 0) {
+		buf = net_buf_alloc(&tx_data_pool_0, K_NO_WAIT);
+	} else {
+		buf = net_buf_alloc(&tx_data_pool_1, K_NO_WAIT);
+	}
+
+	if (buf == NULL) {
+		FAIL("Failed to get buff on ch %i, iteration %i should never happen", chan_idx,
+		     chan_idx);
+	}
+
+	channels[chan_idx].buf = buf;
+	net_buf_reserve(buf, BT_L2CAP_CHAN_SEND_RESERVE);
+	net_buf_add_mem(buf, channels[chan_idx].payload, bytes);
+
+	LOG_DBG("bt_l2cap_chan_sending ch: %i bytes: %i iteration: %i", chan_idx, bytes, iteration);
+	const int ret = bt_l2cap_chan_send(chan, buf);
+
+	LOG_DBG("bt_l2cap_chan_send returned: %i", ret);
+
+	if (ret < 0) {
+		FAIL("Error: send failed error: %i", ret);
+		channels[chan_idx].buf = 0;
+		net_buf_unref(buf);
+	}
+}
+
+void send_sdu_chan_worker(struct k_work *item)
+{
+	const struct channel *ch = CONTAINER_OF(item, struct channel, work);
+
+	send_sdu(ch->iteration, ch->chan_id, ch->bytes_to_send);
+}
+
+static void send_sdu_concurrently(void)
+{
+	for (int i = 0; i < SDU_SEND_COUNT; i++) {
+		for (int k = 0; k < L2CAP_CHANNELS; k++) {
+			channels[k].iteration = i;
+			/* Assign the right msg to the right channel */
+			channels[k].bytes_to_send = (k == LONG_MSG_CHAN_IDX) ? LONG_MSG : SHORT_MSG;
+			const int err = k_work_submit_to_queue(&channels[k].work_queue,
+									&channels[k].work);
+
+			if (err < 0) {
+				FAIL("Failed to submit work to the queque, error: %d", err);
+			}
+		}
+
+		/* Wait until messages on all of the channels has been sent */
+		for (int l = 0; l < L2CAP_CHANNELS; l++) {
+			k_sem_take(&sent_sem, K_FOREVER);
+		}
+	}
+}
+
+static int change_mtu_on_channels(int num_channels, int new_mtu)
+{
+	struct bt_l2cap_chan *reconf_channels[ECRED_CHAN_MAX] = { NULL };
+
+	for (int i = 0; i < num_channels; i++) {
+		reconf_channels[i] = &(&channels[i])->le.chan;
+	}
+
+	return bt_l2cap_ecred_chan_reconfigure(reconf_channels, new_mtu);
+}
+
+static void test_peripheral_main(void)
+{
+	device_sync_init(PERIPHERAL_ID);
+	LOG_DBG("*L2CAP ECRED Peripheral started*");
+	init_workqs();
+	int err = bt_enable(NULL);
+
+	if (err) {
```

Comment:
Here I'd split the declaration and assignment up, like in my earlier comment example